frontend/src/components/Searchbar/Searchbar.tsx

```diff
@@ -58,51 +73,113 @@ export interface SearchbarProps extends Omit<InputProps, 'onChange'> {
   value?: string
 
   /**
-   * Function to be invoked when the value in the searchbar input changes (but
-   * the search button has not been clicked).
+   * Optional. Function to be invoked when the value in the searchbar input
+   * changes (but the search button has not been clicked). Note: at least one of
+   * onSearch and onChange must be provided for the searchbar to work properly!
+   * @param newValue value of the search input
    */
   onChange?: (newValue: string) => void
+
+  /**
+   * Optional. If provided, adds a filter button to the searchbar. The initial
+   * (default) value of the filter, which is most likely the option for "No filter".
+   */
+  filterValue?: string
+
+  /**
+   * Optional. The remainder of the filter options, not including the default
+   * filterValue. Will be ignored unless filterValue has been provided.
+   */
+  filterOptions?: string[]
+
+  /**
+   * Optional. Function to be invoked when a filter option has been selected.
+   * Will be ignored unless filterValue has been provided.
+   * @param filterValue the option that was selected
+   */
+  onFilter?: (filterValue: string) => void
 }
 
 export const Searchbar = forwardRef<SearchbarProps, 'input'>(
   (
     {
       onSearch,
       isExpandable = true,
-      isExpanded = false,
+      isExpanded: isExpandedProp = false,
       onExpandIconClick: onExpandIconClickProp,
       onCollapseIconClick: onCollapseIconClickProp,
       value: valueProp,
       onChange: onChangeProp,
       isDisabled,
+      filterValue = '',
+      filterOptions = [],
+      onFilter,
       ...props
     }: SearchbarProps,
     ref,
   ) => {
-    const [value, setValue] = useState<string | undefined>(valueProp)
-
-    const [innerIsExpanded, setInnerIsExpanded] = useState<boolean>(
-      isExpandable ? isExpanded : true,
+    const isMobile = useIsMobile()
+    const [value, setValue] = useState<string>(valueProp ?? '')
+    const [isExpanded, setIsExpanded] = useState(
+      !isExpandable || isExpandedProp,
     )
+    const [filter, setFilter] = useState<string>(filterValue)
+    const [focus, setFocus] = useState<boolean>(false)
+
+    const { isOpen, onClose, onOpen } = useDisclosure()
 
     const styles = useMultiStyleConfig(SEARCHBAR_THEME_KEY, {
-      isExpanded: innerIsExpanded,
+      isExpanded,
       isDisabled,
       ...props,
     })
 
+    const mobileDrawerExtraButtonProps: Partial<ButtonProps> = useMemo(
+      () => ({
+        isFullWidth: true,
+        justifyContent: 'flex-start',
+        variant: 'clear',
+        colorScheme: 'secondary',
+        textStyle: 'body-1',
+      }),
+      [],
+    )
+
+    const filterActive = useMemo(
```

Comment:
nit: in general we name booleans or functions which return booleans starting with `is` or `has`, e.g. `isFilterActive`

Reply:
[72a9d92](https://github.com/opengovsg/FormSG/pull/5458/commits/72a9d92bca2923427e1c5f8dbf187bf4b5bdb538)